src/libutil/file-system.cc

```diff
@@ -331,15 +331,16 @@ void syncParent(const Path & path)
 
 void recursiveSync(const Path & path)
 {
-    /* If it's a file, just fsync and return. */
+    /* If it's a file or symlink, just fsync and return. */
```

Comment:
nit: this comment is wrong, if I'm reading the code correctly:

If it's a file, fsync and return; if it's a symlink, just return without fsyncing.